vhdl_lang/src/syntax/disconnection.rs

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2018, Olof Kraigher olof.kraigher@gmail.com
+
+use super::common::ParseResult;
+use super::expression::parse_expression;
+use super::names::parse_name;
+use super::separated_list::parse_ident_list;
+use super::tokens::{Kind::*, TokenSpan};
+use crate::ast::token_range::WithTokenSpan;
+use crate::ast::*;
+use vhdl_lang::syntax::parser::ParsingContext;
+
+/// LRM 7.4 Disconnection Specification
+pub fn parse_disconnection_specification(
+    ctx: &mut ParsingContext<'_>,
+) -> ParseResult<WithTokenSpan<DisconnectionSpecification>> {
+    let start_token = ctx.stream.expect_kind(Disconnect)?;
+    if ctx.stream.next_kind_is(Identifier) {
+        let _ = parse_ident_list(ctx);
+    } else if ctx.stream.next_kind_is(All) {
+        ctx.stream.expect_kind(All)?;
+    } else {
+        ctx.stream.expect_kind(Others)?;
+    }
+    ctx.stream.expect_kind(Colon)?;
+    let _ = parse_name(ctx);
+    ctx.stream.expect_kind(After)?;
+    let _ = parse_expression(ctx);
+    let end_token = ctx.stream.expect_kind(SemiColon)?;
+    Ok(WithTokenSpan::new(
+        DisconnectionSpecification {},
+        TokenSpan::new(start_token, end_token),
+    ))
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
These tests are very good and exactly what we want. They check for correctness and, more importantly, avoid regressions.
In addition to these tests, you could optionally also test cases that are expected to fail, i.e.
```vhdl
disconnect S: T before 42 ms;
```

Generally, I think that this is not super important here but this helps in justifying that we show good diagnostics, and that we don't let code pass that is clearly wrong.

Reply:
Done. I have added a few more test cases that fail. I had a little problem, because some of these error-test-cases have no diagnostics. In these cases I simply asserted if the declarations had any errors.

Reply:
Nice!